echemdb/data/cv/entry.py

```diff
@@ -175,197 +190,148 @@ def names(_, context, role):
         from astropy import units as u
 
-        if xunit is None:
-            xunit = u.V
-        if xunit == 'original':
-            xunit = self.figure_description.axes.E.unit
-
-        return u.Unit(xunit)
-
-    def y_unit(self, yunit=None):
-        r"""
-        Return the unit of the y-axis as an astropy unit.
-
-        EXAMPLES:
-
-        Without parameters, SI units are returned::
-
-            >>> entry = Entry.create_examples()[0]
-            >>> entry.y_unit()
-            Unit("A / m2")
-
-        When set to `"original"`, the original units of the published figure are returned::
-
-            >>> entry.y_unit(yunit='original')
-            Unit("mA / cm2")
-
-        Units can be specified explicitly::
+        package = deepcopy(self.package)
+        fields = self.package.get_resource('echemdb').schema.fields
+        df = self.df.copy()
 
-            >>> entry = Entry.create_examples()[0]
-            >>> entry.y_unit(yunit='uA / cm2')
-            Unit("uA / cm2")
+        for idx, field in enumerate(fields):
+            if field.name in new_units.keys():
+                df[field.name] *= u.Unit(field['unit']).to(
+                    u.Unit(new_units[field.name])
+                )
+                package.get_resource('echemdb')["schema"]["fields"][idx][
+                    "unit"
+                ] = new_units[field["name"]]
 
-        """
-        from astropy import units as u
-
-        if yunit is None:
-            if self.y() == 'j':
-                yunit = u.A / u.m**2
-            elif self.y() == 'I':
-                yunit = u.A
-            else:
-                raise NotImplementedError("Unexpected naming of y axis.")
+        package.get_resource('echemdb').data = df.to_csv(index=False).encode()
 
-        if yunit == 'original':
-            yunit = self.figure_description.axes.j.unit or self.figure_description.axes.I.unit
+        return Entry(package=package, bibliography=self.bibliography)
 
-        return u.Unit(yunit)
-
-    def df(self, xunit=None, yunit=None):
+    @property
+    def df(self):
         r"""
-        Return the CSV resource attached to this entry as a data frame.
-
-        If the x and y-units are not specified, all values
-        are in SI units. The data frame can also be returned 
-        with the original figure units or with custom units as shown 
-        in the following examples.
+        Return the CSV resource attached to this entry as a data frame,
+        based on the units given in `entry.data_description.fields`.
 
         EXAMPLES:
-        
-        A data frame in SI units::
-
-            >>> entry = Entry.create_examples()[0]
-            >>> entry.df()
-                         t         E         j
-            0     0.000000 -0.103158 -0.998277
-            1     0.100000 -0.098158 -0.916644
-            ...
 
-        A data frame in the original units of the figure::
+        A data frame of the CSV resource::
 
```

Comment:
```suggestion
```

I don't think this comment helps to understand what is going on. (I think that most readers are going to be confused when they read about the "CSV resource" as they are non-experts in frictionless lingo.)